From f0b48e031d84feeace2313625f02887277759d71 Mon Sep 17 00:00:00 2001
From: Geert Uytterhoeven <geert+renesas@glider.be>
Date: Thu, 6 Aug 2026 12:06:43 +0200
Subject: [PATCH 11236/15333] wifi: nxp: NXPWIFI should be invisible and
 selected by its users

All supported NXP WiFi wireless adapters have an SDIO interface.
Hence there is no point in asking the user about these adapters when
configuring a kernel without MMC support.

Fix this by making the core driver symbol invisible, and selecting it by
its user when needed.

Signed-off-by: Geert Uytterhoeven <geert+renesas@glider.be>
Link: https://patch.msgid.link/aefb37d8398175cb2fb520cb5f725a85bcd3049d.1786010763.git.geert+renesas@glider.be
Signed-off-by: Johannes Berg <johannes.berg@intel.com>
---
 drivers/net/wireless/nxp/nxpwifi/Kconfig | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/net/wireless/nxp/nxpwifi/Kconfig b/drivers/net/wireless/nxp/nxpwifi/Kconfig
index 3637068574b8..38a8820084dd 100644
--- a/drivers/net/wireless/nxp/nxpwifi/Kconfig
+++ b/drivers/net/wireless/nxp/nxpwifi/Kconfig
@@ -1,6 +1,6 @@
 # SPDX-License-Identifier: GPL-2.0-only
 config NXPWIFI
-	tristate "NXP WiFi Driver"
+	tristate "NXP WiFi Driver" if COMPILE_TEST
 	depends on CFG80211
 	help
 	  This adds support for wireless adapters based on NXP
@@ -11,7 +11,8 @@ config NXPWIFI
 
 config NXPWIFI_SDIO
 	tristate "NXP WiFi Driver for IW61x"
-	depends on NXPWIFI && MMC
+	depends on CFG80211 && MMC
+	select NXPWIFI
 	select FW_LOADER
 	select WANT_DEV_COREDUMP
 	help
-- 
2.55.0

